Shopify/tophat
# Improve window activation and management

### What does this change accomplish?

This change resolves a number of issues with window activation, including an issue where the Settings window would open behind other applications.

Resolves #9

### How have you achieved it?

- By adding a consistent call to `NSRunningApplication.current.activate()` when pressing a menu item, which is more reliable than the previous `NSApp.activate(ignoringOtherApps: true)` approach (which is also deprecated).
- By adding a familiar "blink" animation when a menu item is selected, for fun.
- By automatically showing and hiding the Tophat icon in the dock when certain windows are open, so that they're easier to get to.
- `onTrackingHover` has been removed because it seems like some of its previous bugs were fixed, and the remaining ones are acceptable.  The stock `onHover` behaves a bit more reliably in general.

### How can the change be tested?

1. Build and run Tophat.
2. Activate another application.  Ensure that opening Settings from the Tophat menu activates the application and brings the settings window to the front.
3. Ensure that other menu items that open windows continue to activate the application and bring windows to the front as before.
4. Ensure that the blink animation is clean.
5. Ensure that while opening the Settings, About, and Welcome windows, the Dock icon appears, and that clicking on it reveals them.  No other windows should make the Dock icon appear.
